crypto/bn/bn_prime.pl

```diff
@@ -95,12 +95,58 @@
 printf "# define NUMPRIMES %d\n",$eight;
 printf "typedef unsigned char prime_t;\n";
 printf "#endif\n";
-print "static const prime_t primes[NUMPRIMES]= {\n    ";
+print "static const prime_t primes[NUMPRIMES] = {\n    ";
 $init=0;
 for ($i=0; $i <= $#primes; $i++)
-	{
-	printf "\n#ifndef EIGHT_BIT\n    " if ($primes[$i] > 256) && !($init++);
-	printf "\n    " if (($i%8) == 0) && ($i != 0);
-	printf "%4d, ", $primes[$i];
-	}
+    {
+    printf "\n#ifndef EIGHT_BIT" if ($primes[$i] > 256) && !($init++);
+    printf "\n    " if (($i%9) == 0) && ($i != 0);
+    printf "%6d, ", $primes[$i];
+    }
 print "\n#endif\n};\n";
+
+print `echo "
+
+primes = [2, 3, 5, 7, 11, 13]
+
+safe_prefix = 'safe_'
+
+for safe in (False, True):
+    muliplier = 1 if not safe else 2
+    for p in primes:
+        muliplier *= p
+
+    offsets = []
+    for x in range(3, muliplier + 3, 2):
+        prime = True
+        for p in primes:
+            if not x % p or (safe and not ((x - 1) / 2) % p):
+                prime = False
+                break
+
+        if prime:
+            offsets.append(x)
+
+    print('#define %sPRIME_OFFSET_COUNT %d' % (
+        safe_prefix.upper() if safe else '', len(offsets)))
+    print('static const uint %sprime_offsets[%sPRIME_OFFSET_COUNT] = {' % (
+        safe_prefix if safe else '',
+        safe_prefix.upper() if safe else ''))
+    print_buffer = '    '
+    for offset in offsets:
+        if len(print_buffer) > 60:
+            print(print_buffer)
+            print_buffer = '    '
+        print_buffer += '%6d,' % offset
+    print(print_buffer)
+    print('};')
+    print('static const uint %sprime_multiplier = %d;' % (
+        safe_prefix if safe else '', muliplier))
+    print('static const int %sprime_multiplier_bits = %d;' % (
+        safe_prefix if safe else '', muliplier.bit_length()))
+
+    print('')
+
+print('static const int first_prime_index = %d;' % len(primes))
+
+" | python`;
```

Comment:
I'm inclined to agree, and I'm sure it could, but I'm not at all familiar with perl, and don't have time to learn it right now.